Objects/bytes_methods.c

```diff
@@ -557,10 +557,49 @@ find_internal(const char *str, Py_ssize_t len,
 }
 
 PyObject *
-_Py_bytes_find(const char *str, Py_ssize_t len, PyObject *sub,
+_Py_bytes_find(const char *str, Py_ssize_t len, PyObject *subobj,
                Py_ssize_t start, Py_ssize_t end)
 {
-    Py_ssize_t result = find_internal(str, len, "find", sub, start, end, +1);
+    Py_ssize_t result;
+    if (PyTuple_Check(subobj)) {
+        result = -1;
+        ADJUST_INDICES(start, end, len);
+        // Work in batches of 10000
+        for (; result == -1 && start <= end; start += 10000) {
+            for (Py_ssize_t i = 0; i < PyTuple_GET_SIZE(subobj); i++) {
+                PyObject *subseq = PyTuple_GET_ITEM(subobj, i);
+                Py_ssize_t sublen;
+                Py_buffer subbuf;
+                if (!PyObject_CheckBuffer(subseq)) {
```

Comment:
Could factor out checks before main loop.

Reply:
This is not a check, I need to determine the length of the substring and integers must also be accepted.

Reply:
`PyObject_CheckBuffer` is. If len > 10000, it will be repeated more than once on the same substring.

Reply:
Nope. You can do `b"0".find(0x30)`, if it's not a buffer I assume it's an integer so a length of 1, otherwise I determine the length of the buffer.